lib/src/doc/allow.rs

```diff
@@ -14,30 +14,33 @@ impl<'a> Document<'a> {
 		txn: &Transaction,
 		stm: &Statement<'_>,
 	) -> Result<(), Error> {
-		// Check permission clause
-		if opt.perms && opt.auth.perms() && self.id.is_some() {
-			// Get the table
-			let tb = self.tb(opt, txn).await?;
-			// Get the permission clause
-			let perms = if stm.is_delete() {
-				&tb.permissions.delete
-			} else if stm.is_select() {
-				&tb.permissions.select
-			} else if self.is_new() {
-				&tb.permissions.create
-			} else {
-				&tb.permissions.update
-			};
-			// Match the permission clause
-			match perms {
-				Permission::None => return Err(Error::Ignore),
-				Permission::Full => return Ok(()),
-				Permission::Specific(e) => {
-					// Disable permissions
-					let opt = &opt.perms(false);
-					// Process the PERMISSION clause
-					if !e.compute(ctx, opt, txn, Some(&self.current)).await?.is_truthy() {
-						return Err(Error::Ignore);
+		// Check if this record exists
+		if self.id.is_some() {
+			// Should we run permissions checks?
+			if opt.perms && opt.auth.perms() {
```

Comment:
If we are creating a record such that it doesn't already exist, would this bypass `CREATE` permissions checks?

Reply:
The document `id` field is none if the record is not an actual record. For instance with the following query, we are not actually selecting from a record, therefore we will never check table/field permissions checks for the 'documents'...

```sql
SELECT * FROM [1,2,3];
```

With regards to the double `self.allow(...)` call, this is to check (if we are dealing with an actual table record) whether we are allowed to write to the table in the first place, but also whether we are allowed to write to the table with the record contents once it has been set. For instance:
1. If the permissions allow us to update a record
2. But we can only update fields where the `allowed_to_update` field is true.
3. We must first check before editing the document to see if the permissions allow us to edit the document.
4. But we must also check after editing the document to see if the permissions allow us to edit documents with values that we have just changed it to.

Either way, this behaviour hasn't changed. Just that we were looping over every field (and cloning all of the `Idiom` values for every field) on every document, even if we weren't logged in as a `scope` user, and therefore didn't need to process permissions.

Reply:
That makes much more sense, thanks!